docs/codeql/writing-codeql-queries/debugging-data-flow-queries-using-partial-flow.rst

```diff
@@ -0,0 +1,116 @@
+
+To avoid these problems, a data-flow ``Configuration`` comes with a mechanism for exploring partial flow that tries to deal with these caveats. This is the ``Configuration.hasPartialFlow`` predicate:
+
+.. code-block:: ql
+
+      /**
+       * Holds if there is a partial data flow path from `source` to `node`. The
+       * approximate distance between `node` and the closest source is `dist` and
+       * is restricted to be less than or equal to `explorationLimit()`. This
+       * predicate completely disregards sink definitions.
+       *
+       * This predicate is intended for dataflow exploration and debugging and may
+       * perform poorly if the number of sources is too big and/or the exploration
+       * limit is set too high without using barriers.
+       *
+       * This predicate is disabled (has no results) by default. Override
+       * `explorationLimit()` with a suitable number to enable this predicate.
+       *
+       * To use this in a `path-problem` query, import the module `PartialPathGraph`.
+       */
+      final predicate hasPartialFlow(PartialPathNode source, PartialPathNode node, int dist) {
+
+As noted in the documentation for ``hasPartialFlow`` (for example, in the `CodeQL for Java documentation <https://codeql.github.com/codeql-standard-libraries/java/semmle/code/java/dataflow/internal/DataFlowImpl2.qll/predicate.DataFlowImpl2$Configuration$hasPartialFlow.3.html>__`) you must first enable this by adding an override of ``explorationLimit``. For example:
+
+.. code-block:: ql
+
+    override int explorationLimit() { result = 5 }
+
+This defines the exploration radius within which ``hasPartialFlow`` returns results.
+
+It is also useful to focus on a single ``source`` at a time as the starting point for the flow exploration. This is most easily done by adding a temporary restriction in the ``isSource`` predicate.
+
+To do quick evaluations of partial flow it is often easiest to add a predicate to the query that is solely intended for quick evaluation (right-click the predicate name and choose "CodeQL: Quick Evaluation"). A good starting point is something like:
+
+.. code-block:: ql
+
+    predicate adhocPartialFlow(Callable c, PartialPathNode n, Node src, int dist) {
+      exists(MyConfig conf, PartialPathNode source |
+        conf.hasPartialFlow(source, n, dist) and
+        src = source.getNode() and
+        c = n.getNode().getEnclosingCallable()
+      )
+    }
+
+If you are focusing on a single ``source`` then the ``src`` column is meaningless. You may of course also add other columns of interest based on ``n``, but including the enclosing callable and the distance to the source at the very least is generally recommended, as they can be useful columns to sort on to better inspect the results.
```

Comment:
This was rewritten from "superfluous" to "meaningless", which I don't think makes sense. It should be reverted to "superfluous".